cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst

```diff
@@ -0,0 +1,134 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :description: Cask Data Application Platform Wikipedia Pipeline Application
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _examples-wikipedia-data-pipeline:
+
+==================
+Wikipedia Pipeline
+==================
+
+A Cask Data Application Platform (CDAP) example demonstrating a typical data processing pipeline using CDAP Workflows.
+
+Overview
+========
+
+This example demonstrates a CDAP application performing analysis on Wikipedia data using MapReduce and Spark programs
+running within a CDAP Workflow - *WikipediaPipelineWorkflow*.
+
+This example can be run in both online and offline modes. In the online mode, the custom action *DownloadWikiDataAction* reads the *pageTitleStream*
+in which each event is one element from the output of the facebook likes API - https://developers.facebook.com/docs/graph-api/reference/v2.4/object/likes
+For each event, it tries to download Wikipedia data for the page using the https://www.mediawiki.org/wiki/API:Main_page API.
+It stores the downloaded data in the KeyValueTable dataset *wikiData*.
+In the offline mode, it expects Wikipedia data formatted per the output of the MediaWiki API above in the stream *wikiStream*.
+The MapReduce program *wikiDataToDataset* consumes this stream and stores it in the same KeyValueTable dataset *wikiData*.
+Data can be uploaded to the *wikiStream* using the CDAP CLI.
+
+Once raw Wikipedia data is available using in either online or offline modes, the *WikipediaPipelineWorkflow* runs a MapReduce program
+*WikiContentValidatorAndNormalizer* that filters bad records from the raw data, as well as normalizes it by converting the wikitext-formatted
+data to plain text. It then stores the output in another KeyValueTable dataset *normalized*
+
+The *WikipediaPipelineWorkflow* then contains a fork, with two branches in parallel. One branch runs the Apache Spark program *SparkWikipediaAnalyzer*.
+This program consumes normalized data and runs topic modeling on it using the Latent Dirichlet Allocation (LDA) algorithm. It stores its

```

Comment:
Out of the multiclass classifiers that MLLib provides, Naive Bayes would probably work best for this problem. Do we care about the accuracy?